sys/cheri/cheric.h

```diff
@@ -272,26 +272,14 @@ cheri_bytes_remaining(const void * __capability cap)
 
 #define CHERI_FPRINT_PTR(f, ptr)					\
 	fprintf(f, _CHERI_PRINT_PTR_FMT(ptr))
-#endif /* __has_feature(capabilities) || defined(__CHERI__) */
 
-/* Allow use of some cheri_ptr macros in the purecap kernel
- * without extra ifdefs.
- * These become a no-op when compiling an hybrid kernel or a
- * normal kernel.
- */
-#ifdef CHERI_PURECAP_KERNEL
-
-#define cheri_bound(ptr, size) cheri_ptr((const void *)(ptr), size)
-#define cheri_perm(ptr, size, perm) cheri_ptrperm((const void *)(ptr), size, perm)
-#define cheri_valid(ptr) (cheri_gettag((const void *)(ptr)) == 1)
-
-#else /* ! CHERI_PURECAP_KERNEL */
-
-#define cheri_bound(ptr, size) (ptr)
-#define cheri_perm(ptr, size, perm) (ptr)
-#define cheri_valid(ptr) (1)
-
-#endif /* ! CHERI_PURECAP_KERNEL */
+#else /* ! (__has_feature(capabilities) || defined(__CHERI__)) */
```

Comment:
We use some of the builtins in the hybrid kernel though, so I'm a bit leery of redefining the stock builtins.

Reply:
Good point, with the hybrid kernel gets quite messy, I originally had them with different names and then removed those in a cleanup commit. Do you have a suggestion for an appropriate prefix? I was thinking `purecap_setbounds` or `cheri_kern_setbounds`.

Reply:
The latter seems sensible if we don't want to got all the ways to `setbounds`